util.c

```diff
@@ -211,3 +211,67 @@ int r_mkdir(char *path)
 	return 0;
 }
 
+void render_core(int sx, int sy, int sw, int sh, int dx, int dy, int dw, int dh,
+                 unsigned long bgc, bool alpha) {
```

Comment:
I moved it because it is also used in thumbs.c, and because all the functions exported from image.c take an `img_t*` argument, so I didn't think it fit in. But I can just as well move it back to image.c if you think it's more suitable.

Reply:
>image.c take an img_t* argument,  

You could have render_core do the same. Right now the code just assumes the image is the current context instead of having the caller explicitly pass it in.
Also do we need to specify alpha or can we just infer it from the img_t?

Reply:
> What happens when thumbnails are rendered via `imlib_render_image_on_drawable_at_size` instead of `render_core` ?

This happens, i.e. parts of the image that should be opaque also become transparent.
![thumbs](https://user-images.githubusercontent.com/6820152/136258196-3ba70477-54ab-48ab-9d28-f26d04cb2896.png)

I think it's because `imlib_render_image_on_drawable_at_size` doesn't merge the alpha channel of the image onto the drawable. The function `imlib_blend_image_onto_image` which I use in `render_core` takes an argument `merge_alpha`, which enables this. The `render_core` function is also needed to get premultiplied alpha correctly.

Reply:
> > image.c take an img_t* argument,
> 
> You could have render_core do the same. Right now the code just assumes the image is the current context instead of having the caller explicitly pass it in. Also do we need to specify alpha or can we just infer it from the img_t?

The problem is that there's no img_t available when rendering thumbnails. We could pass in an Imlib_Image instead of taking it from the context, but I don't see the benefit of that. It wouldn't give us alpha either.

Reply:
> This happens, i.e. parts of the image that should be opaque also become transparent

I see, was hoping it would be something less catastrophic.